cli/command/cli_options.go

```diff
@@ -11,19 +11,11 @@ import (
 	"github.com/moby/term"
 )
 
-// CLIOption applies a modification on a DockerCli.
+// CLIOption is a functional argument to apply options to a [DockerCli]. These
+// options can be passed to [NewDockerCli] to initialize a new CLI, or
+// applied with [DockerCli.Initialize] or [DockerCli.Apply].
```

Comment:
Just for clarity, could It make sense to mention here that it replaces the deprecated DockerCliOption/InitializeOpt types?

Reply:
I don't think we've done that before. We usually;

- Deprecate in 1 release, and add aliases to the new location (to give users a pointer where to look for the new thing)
- Remove in the release after.

Admitted; if we're gonna ramp up release cadence, then 1 release "deprecated' may potentially be too short. OTOH; if we move to Go modules, every major release is ... a major release, so technically "anything goes!"